compiler-core/src/javascript/tests/singleton_comparison_optimisation.rs

```diff
@@ -0,0 +1,120 @@
+use crate::assert_js;
```

Comment:
Ah! I this this new module isn't part of the codebase, so these tests never run.

Could you move these tests to the javascript/test/custom_types.rs module please

Reply:
This one gets me all the time! Such a Rust footgun